File: src/lib/embeds.ts
import { EmbedBuilder } from "discord.js";

/** Brand color — teal/cyan accent matching the dark Discord theme */
export const BRAND_COLOR = 0x2dd4bf;
export const SUCCESS_COLOR = 0x22c55e;
export const ERROR_COLOR = 0xef4444;
export const WARNING_COLOR = 0xf59e0b;

export function brandEmbed() {
  return new EmbedBuilder().setColor(BRAND_COLOR).setTimestamp();
}

export function successEmbed() {
  return new EmbedBuilder().setColor(SUCCESS_COLOR).setTimestamp();
}

export function errorEmbed(message: string) {
  return new EmbedBuilder()
    .setColor(ERROR_COLOR)
    .setDescription(`${message}`)
    .setTimestamp();
}

File: src/embeds.rs
use poise::serenity_prelude as serenity;

pub const BRAND_COLOR: u32 = 0x2D_D4_BF;
pub const SUCCESS_COLOR: u32 = 0x22_C5_5E;
pub const ERROR_COLOR: u32 = 0xEF_44_44;

pub fn brand() -> serenity::CreateEmbed {
    base(BRAND_COLOR)
}

pub fn success() -> serenity::CreateEmbed {
    base(SUCCESS_COLOR)
}

pub fn error(message: impl Into<String>) -> serenity::CreateEmbed {
    base(ERROR_COLOR).description(message)
}

pub fn log(title: impl Into<String>, description: impl Into<String>) -> serenity::CreateEmbed {
    brand().title(title).description(description)
}

fn base(color: u32) -> serenity::CreateEmbed {
    serenity::CreateEmbed::new()
        .color(serenity::Colour::new(color))
        .timestamp(serenity::Timestamp::now())
}
